diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt

```diff
@@ -135,11 +122,79 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
             else -> {
             }
         }
     }
 
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION", "MagicNumber")
+    private fun handDotQualifiedAndSafeAccessExpression(node: ASTNode) {
+        val listParentTypesNoFix = listOf<IElementType>(PACKAGE_DIRECTIVE, IMPORT_DIRECTIVE, VALUE_PARAMETER_LIST,
+            VALUE_ARGUMENT_LIST, DOT_QUALIFIED_EXPRESSION, SAFE_ACCESS_EXPRESSION, POSTFIX_EXPRESSION)
+        if (isNotFindParentNodeWithSpecificManyType(node, listParentTypesNoFix)) {
+            val listDot: MutableList<ASTNode> = mutableListOf()
+            searchDot(node, listDot)
+            if (listDot.size > 3) {
+                val (withRightWhiteSpace, without) = listDot.partition {
+                    val whiteSpaceBeforeDotOrSafeAccess = it.findChildByType(DOT)?.treePrev ?: it.findChildByType(SAFE_ACCESS)?.treePrev
+                    whiteSpaceBeforeDotOrSafeAccess?.elementType == WHITE_SPACE && whiteSpaceBeforeDotOrSafeAccess.text.lines().size > 1
+                }
+                if (without.size > 1 || (without.size == 1 && without[0] != listDot[0])) {
+                    WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode, "should be split before second and other dot/safe access", node.startOffset, node) {
+                        fixDotQualifiedExpression(listDot)
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Return false, if you find parent with types in list else return true
+     */
+    private fun isNotFindParentNodeWithSpecificManyType(node: ASTNode, list: List<IElementType>): Boolean {
+        list.forEach { elem ->
+            node.findParentNodeWithSpecificType(elem)?.let {
+                return false
+            }
+        }
+        return true
+    }
+
+    /**
+     * Fix Dot Qualified Expression and Safe Access Expression -
+     * 1) Append new White Space node before second and subsequent node Dot or Safe Access
+     * in Dot Qualified Expression? Safe Access Expression and Postfix Expression
+     * 2) If before first Dot or Safe Access node stay White Space node with \n - remove this node
+     */
+    private fun fixDotQualifiedExpression(list: MutableList<ASTNode>) {
+        list.forEachIndexed { index, astNode ->
+            val dotNode = astNode.getFirstChildWithType(DOT) ?: astNode.getFirstChildWithType(SAFE_ACCESS)
+            val nodeBeforeDot = dotNode?.treePrev
+            if (index > 0) {
+                astNode.appendNewlineMergingWhiteSpace(nodeBeforeDot, dotNode)
+            }
+        }
+    }
+
+    private fun isDotQuaOrSafeAccessOrPostfixExpression(node: ASTNode) =
+            node.elementType == DOT_QUALIFIED_EXPRESSION || node.elementType == SAFE_ACCESS_EXPRESSION || node.elementType == POSTFIX_EXPRESSION
+
+    private fun searchDot(node: ASTNode, dotList: MutableList<ASTNode>) {
```

Comment:
we have function `org.cqfn.diktat.ruleset.utils.AstNodeUtilsKt#findAllNodesWithCondition` and it can be reused:
```kotlin
val listDot = node.findAllNodesWithCondition(
    withSelf = true,
    excludeChildrenCondition = { !isDotQuaOrSafeAccessOrPostfixExpression(it) }
) {
    isDotQuaOrSafeAccessOrPostfixExpression(it) && it.elementType != POSTFIX_EXPRESSION
}.reversed()
```